pkg/nodeclaim/nodeclaim.go

```diff
@@ -250,3 +320,18 @@ func CheckNodeClaimStatus(ctx context.Context, nodeClaimObj *v1beta1.NodeClaim,
 		}
 	}
 }
+
+func IsNodeClassAvailable(ctx context.Context, kubeClient client.Client) bool {
```

Comment:
This client is controller-runtime client. If you don't watch the nodeclaim object, every GET will become a direct API call. I suggest you move this code out of reconciler (to main function, just execute once when controller starts). The nodeclass is not associated with any workspace. It is weird to see it is reconciled whenever we reconcile a workspace.